src/main/java/auth/validator/UserValidator.java

```diff
@@ -0,0 +1,49 @@
+package auth.validator;
+
+import org.apache.commons.validator.routines.EmailValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+import auth.model.User;
+import auth.service.UserService;
+
+@Component
+public class UserValidator implements Validator {
+
+	@Autowired
+	private UserService userService;
+
+	@Override
+	public boolean supports(Class<?> clazz) {
+		return User.class.equals(clazz);
+	}
+
+	@Override
+	public void validate(Object target, Errors errors) {
+		User user = (User) target;
+		boolean browserValidation = true;
+
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "email", "NotEmpty");
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "password", "NotEmpty");
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "passwordConfirm", "NotEmpty");
+
+		if (!user.getPassword().equals(user.getPasswordConfirm())) {
+			errors.rejectValue("passwordConfirm", "Diff.passwordConfirm");
+			browserValidation = false;
+		}
+
+		if (!EmailValidator.getInstance().isValid(user.getEmail())) {
+			errors.rejectValue("email", "Email.not.correct");
+			browserValidation = false;
+		}
+
+		if (browserValidation && userService.findUserByEmail(user.getEmail()) != null) {
```

Comment:
Hmm. can't you just test if there are any errors already added? I think that if fields are not fillet still the db call is done.